draft-ietf-tls-esni.md

```diff
@@ -792,50 +780,21 @@ the "server_name" extension. Any actual "server_name" extension is
-the TLS connection to the backend server (if in Split Mode). In
-the latter case, it does not make any changes to the TLS
-messages, but just blindly forwards them.
-
-If the ClientHello is the result of a HelloRetryRequest, servers MUST
-abort the connection with an "illegal_parameter" alert if any of the
-ClientEncryptedSNI.suite, ClientEncryptedSNI.key_share, ClientEncryptedSNI.record_digest,
-or decrypted ClientESNIInner values from the second ClientHello do not
-match that of the first ClientHello.
-
-## Shared Mode Server Behavior
-
-A server operating in Shared Mode uses PaddedServerNameList.sni as
-if it were the "server_name" extension to finish the handshake. It
-SHOULD pad the Certificate message, via padding at the record layer,
-such that its length equals the size of the largest possible Certificate
-(message) covered by the same ESNI key. Moreover, the server MUST
-include the "encrypted_server_name" extension in EncryptedExtensions
-with the "response_type" field set to "esni_accept" and the "nonce"
-field set to the decrypted PaddedServerNameList.nonce value from the client
-"encrypted_server_name" extension.
-
-If the server sends a NewSessionTicket message, the corresponding ESNI PSK MUST
-be ignored by all other servers in the deployment when not negotiating ESNI,
-including servers which do not implement this specification.
-
-This restriction provides robustness for rollbacks (see {{misconfiguration}}).
-
-## Split Mode Server Behavior {#backend-server-behavior}
-
-In Split Mode, the backend server must know PaddedServerNameList.nonce
-to echo it back in EncryptedExtensions and complete the handshake.
-{{communicating-sni}} describes one mechanism for sending both
-PaddedServerNameList.sni and ClientESNIInner.nonce to the backend
-server. Thus, backend servers function the same as servers operating
-in Shared Mode.
-
-As in Shared Mode, if the backend server sends a NewSessionTicket message, the
-corresponding ESNI PSK MUST be ignored by other servers in the deployment when
-not negotiating ESNI, including servers which do not implement this
-specification.
+Upon determining the true SNI, the client-facing server inverts the
+transformation of the ClientHello as defined in {{esni-transformation}}, then
+forwards the restored message to the backend server.  The inverse transformation
+MUST also be applied to the ClientHello that the client sends in in response to
+a HelloRetryRequest.
+
+## Backend Server Behavior {#backend-server-behavior}
+
+The backend server SHOULD pad the Certificate message, via padding at the record
+layer, such that its length equals the size of the largest possible Certificate
+message covered by the same ESNI keys.
+
+The backend server SHOULD ignore the "encrypted_server_name" extension of type
+`esni_parameters`, as the purpose of this parameters is to derive the handshake
+traffic secret from a handshake transcript invisible to an observer.
 
```

Comment:
Really? If we do he QUIC authenticated handshake, are we not going to use for example the nonce as a way to authenticate the QUIC initial packets?

Reply:
Fair point. 2586cd2 points out that the parameters might be used by future extensions.